crates/router/src/core/payments.rs

```diff
@@ -99,27 +99,35 @@ where
 
     payment_data.payment_method_data = payment_method_data;
 
-    let connector_details = operation
+    let connector_choice = operation
         .to_domain()?
-        .get_connector(
-            &merchant_account,
-            state,
-            &req,
-            payment_data.payment_attempt.connector.as_ref(),
-        )
+        .get_connector(&merchant_account, state, &req)
```

Comment:
Create a new function and add the below logic and this function in that.

Reply:
Are you sure we want to move `.get_connector()` in a separate function? It's using the `operation` and invoking a domain function.

Reply:
The agenda is core just call a connector function and all logic's abstracted inside that